### PR TITLE
python3Packages.transformers: 4.4.2 -> 4.12.2

### PR DESCRIPTION
**Pull request by rissson**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by rissson** [comment]
Result of `nixpkgs-review pr 144269` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python38Packages.spacy-transformers</li>
    <li>python39Packages.spacy-transformers</li>
  </ul>
</details>
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python38Packages.datasets</li>
    <li>python38Packages.huggingface-hub</li>
    <li>python38Packages.tokenizers</li>
    <li>python38Packages.transformers</li>
    <li>python39Packages.datasets</li>
    <li>python39Packages.huggingface-hub</li>
    <li>python39Packages.tokenizers</li>
    <li>python39Packages.transformers</li>
  </ul>
</details>

**Review by jonringer — COMMENTED** [review]
looks like this breaks `spacy-transformers`, do you mind patching the version range.

```
https://github.com/NixOS/nixpkgs/pull/144269

2 packages failed to build:
python38Packages.spacy-transformers python39Packages.spacy-transformers

8 packages built:
python38Packages.datasets python38Packages.huggingface-hub python38Packages.tokenizers python38Packages.transformers python39Packages.datasets python39Packages.huggingface-hub python39Packages.tokenizers python39Packages.transformers
```

**Comment by rissson** [comment]
> looks like this breaks `spacy-transformers`, do you mind patching the version range.

I just did!

**Comment by rissson** [comment]
Result of `nixpkgs-review pr 144269` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>python38Packages.datasets</li>
    <li>python38Packages.huggingface-hub</li>
    <li>python38Packages.spacy-transformers</li>
    <li>python38Packages.tokenizers</li>
    <li>python38Packages.transformers</li>
    <li>python39Packages.datasets</li>
    <li>python39Packages.huggingface-hub</li>
    <li>python39Packages.spacy-transformers</li>
    <li>python39Packages.tokenizers</li>
    <li>python39Packages.transformers</li>
  </ul>
</details>

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 144269 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>2 packages failed to build:</summary>  <ul>    <li>python38Packages.spacy-transformers</li>    <li>python39Packages.spacy-transformers</li>  </ul></details><details>  <summary>8 packages built:</summary>  <ul>    <li>python38Packages.datasets</li>    <li>python38Packages.huggingface-hub</li>    <li>python38Packages.tokenizers</li>    <li>python38Packages.transformers</li>    <li>python39Packages.datasets</li>    <li>python39Packages.huggingface-hub</li>    <li>python39Packages.tokenizers</li>    <li>python39Packages.transformers</li>  </ul></details>

**Comment by jonringer** [comment]
sigh, script picked up old results

```
https://github.com/NixOS/nixpkgs/pull/144269

10 packages built:
python38Packages.datasets python38Packages.huggingface-hub python38Packages.spacy-transformers python38Packages.tokenizers python38Packages.transformers python39Packages.datasets python39Packages.huggingface-hub python39Packages.spacy-transformers python39Packages.tokenizers python39Packages.transformers
```

File: pkgs/development/python-modules/spacy-transformers/default.nix [hunk 1]
```diff
@@ -21,6 +21,10 @@ buildPythonPackage rec {
     sha256 = "b84c195dc21a28582579dea3f76c90222e29ee0d99b6adf38ade75646ed2746e";
   };
 
+  postPatch = ''
+    sed -i 's/transformers>=3.4.0,<4.12.0/transformers/' setup.cfg
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/spacy-transformers/default.nix:25]: This should use substituteInPlace

Comment (rissson) [reply pkgs/development/python-modules/spacy-transformers/default.nix:25]: Should I open a PR to change that?

**Pull request merged**